### PR TITLE
Added ItemType():getRequiredLevel()

### PR DESCRIPTION
**Pull request by Codinablack**
Added this function for everyone who may need it. Could be very useful for all scripters. I fixed everything, }else{ like was pointed out, and made sure to have the extra line after the new method. I hope finally everything is fine for this pull request to be accepted.

### CODE REVIEW

File: src/luascript.cpp [hunk 1]
```diff
@@ -11550,6 +11551,18 @@ int32_t LuaScriptInterface::luaItemTypeGetDecayId(lua_State* L)
 	return 1;
 }
 
+int32_t LuaScriptInterface::luaItemTypeGetRequiredLevel(lua_State* L)
+{
+	// itemType:getRequireLevel()
+	const ItemType* itemType = getUserdata<const ItemType>(L, 1);
+	if (itemType) {
+		lua_pushnumber(L, itemType->minReqLevel);
+	}else {
```
Comment (att3) [inline src/luascript.cpp]: } else {

Comment (Codinablack) [inline src/luascript.cpp]: Dude, really? Wtf is your problem hopping on my PR's like that? So cry about it, one space is missing, I just messed with git for an hr trying to squash and make everything right, and you come in with no time nedeed to mess it up again. Not happy in life and got to try to bring others down too or something?

Comment (djarek) [inline src/luascript.cpp]: Code formatting is important :). BTW. Easiest way to fix it is to push a new commit and then do rebase -i HEAD~2. Pick the first commit and squash the second, then change the commit message et voila!

Comment (att3) [inline src/luascript.cpp]: @Codinablack 
Don't take it so personally.
I saw a tiny typo you had made and commented it out, not trying bring you down or anything, rather trying to help you.

I feel you with the git issues though, if I had noticed you were having them, I probably wouldn't have commented anything.

Comment (WibbenZ) [inline src/luascript.cpp]: I can agree, I hate using git and I can not make the git software work...
Can't even get it to update here on my repo without it saying I don't own this repo. Idk... tried to google things for weeks with no luck.
But the thing is we should keep using this code format, better then ex. 0.4 where it's all just a big mess (just try to compare reading the protocollogin file or something else that isen't huge)...

**Comment by Codinablack** [comment]
commitsquash

Yeah so easy.. Same fucking problems as before, I do exactly as you guys tell me, and all the links that I am sent to tell me to do. I make sure I'm in the right directory first thing I do, second thing is pull from main branch online to make sure it's up to date, it retrieves my newest commit online, then I squash the commits with rebase, pretty easy, edit the notepad and viola! It squashes the commits, then I try to push to master, what happened in the screenshot? Some how some way I am behind the master online again now, even though I'm the only one with write access and haven't done anything else, now before I take this next screenshot I will try one more time to push, it won't let me, then I will pulll and push and this PR is going to get a lot more commits, watch....

**Comment by Codinablack** [comment]
Ok so I got it figured out, not a single link or direction given by anybody was the solution. Always the same problem, that was because of fast-forwarding, and I got around it by using --force option when pushing again. I hope all is now well for this thing to be accepted.

@kornholi

File: src/luascript.cpp [hunk 2]
```diff
@@ -11550,6 +11551,18 @@ int32_t LuaScriptInterface::luaItemTypeGetDecayId(lua_State* L)
 	return 1;
 }
 
+int32_t LuaScriptInterface::luaItemTypeGetRequiredLevel(lua_State* L)
+{
+	// itemType:getRequireLevel()
```
Comment (dalkon) [inline src/luascript.cpp]: This does not match the function name.

**Comment by Codinablack** [comment]
@dalkon fixed

**Comment by Codinablack** [comment]
Is there anything left for me to do for this to become approved?

**Comment by marksamman** [comment]
Thanks for the contribution!

**Comment by Codinablack** [comment]
Thanks for the approval :)

**Pull request merged**